packages/react-sdk/README.md

```diff
@@ -93,18 +108,20 @@ const optimizely = createInstance({
 })
 ```
 
-
 ## `<OptimizelyProvider>`
 
 Required at the root level. Leverages React’s `Context` API to allow access to the `ReactSDKClient` to components like `<OptimizelyFeature>` and `<OptimizelyExperiment>`.
 
 *props*
 * `optimizely : ReactSDKClient` created from `createInstance`
-* `user: { id: string; attributes?: { [key: string]: any } }` User info object - `id` and `attributes` will be passed to the SDK for every feature flag, A/B test, or `track` call
-* `timeout : Number` (optional) The amount of time for OptimizelyExperiment and OptimizelyFeature components to render `null` before resolving
+* `user: { id: string; attributes?: { [key: string]: any } } | Promise` User info object - `id` and `attributes` will be passed to the SDK for every feature flag, A/B test, or `track` call, or a `Promise` for the same kind of object
+* `timeout : Number` (optional) The amount of time for OptimizelyExperiment and OptimizelyFeature components to render `null` while waiting for the SDK instance to become ready, before resolving..
 * `isServerSide : Boolean` (optional) must pass `true` here for server side rendering
-
-### Load the datafile synchronously
+* `userId : String` (optional) Another way to provide user id. The `user` object prop takes precedence when both are provided.
+* `userAttributes : Object` : (optional) Another way to provide user attributes. The `user` object prop takes precedence when both are provided.
+### Readiness
+Before rendering real content, both the datafile and the user must be available to the SDK.
+#### Load the datafile synchronously
 
 Synchronous loading is the preferred method to ensure that Optimizely is always ready and doesn't add any delay or asynchronous complexity to your application.
```

Comment:
Readers of these docs may ask, "How do I get `window.datafile` defined?" I think we could put a line here that either mentions the tag version of the datafile:

`cdn.optimizely.com/datafiles/123.json/tag.js`

Or it mentions that one should have a server pass the datafile to the client.

Reply:
I'll add that line.